src/main/java/software/aws/neptune/gremlin/GremlinConnectionProperties.java

```diff
@@ -1120,4 +1134,32 @@ protected void validateProperties() throws SQLException {
     public boolean isSupportedProperty(final String name) {
         return SUPPORTED_PROPERTIES_LIST.contains(name);
     }
+
+    /**
+     * Updates the SERVICE_REGION env variable in JVM to the region set by Gremlin connection properties
+     */
+    @SuppressWarnings({ "unchecked" })
+    private static void setRegionEnv(final String region) throws Exception {
+        if (region.equals(System.getenv("SERVICE_REGION"))) {
+            return;
+        }
+        if (System.getenv("SERVICE_REGION") != null) {
+            LOGGER.warn(String.format("Overriding the current SERVICE_REGION environment variable with %s.", region));
+        }
+        try {
```

Comment:
Make into function.
```
    void setEnvironmentVariable(final String key, final String value) throws SQLException {
        try {
            if (SystemUtils.IS_OS_WINDOWS) {
                setWindowsEnvironmentVariable(key, value);
            } else {
                setMacEnvironmentVariable(key, value);
            }
        } catch (final Exception e) {
            throw new SQLException("<Error message>");
        }
    }

    void setWindowsEnvironmentVariable(final String key, final String value) throws Exception {
        final Class<?> processEnv = Class.forName("java.lang.ProcessEnvironment");
        final Method getenv = processEnv.getDeclaredMethod("getenv", String.class);
        getenv.setAccessible(true);
        final Field caseInsensitiveEnv = processEnv.getDeclaredField("theCaseInsensitiveEnvironment");
        caseInsensitiveEnv.setAccessible(true);
        final Map<String, String> envMap = (Map<String, String>) caseInsensitiveEnv.get(null);
        envMap.put(key, value);
    }

    void setMacEnvironmentVariable(final String key, final String value) throws Exception {
        final Map<String, String> env = System.getenv();
        final Field field = env.getClass().getDeclaredField("m");
        field.setAccessible(true);
        ((Map<String, String>) field.get(env)).put(key, value);
    }
```

Reply:
Done, with slight modification to just taking region value and only changing `SERVICE_REGION` in the respective functions.